euth/flashpoll/static/js/flashpoll.js

```diff
@@ -819,3 +826,110 @@ window.changeType = function(questionType, key, qorderId) {
     }
   }
 }
+
+function JSONToCSVConvertor(JSONData, ReportTitle, ShowLabel) {
+    //If JSONData is not an object then JSON.parse will parse the JSON string in an Object
+    var arrData = typeof JSONData != 'object' ? JSON.parse(JSONData) : JSONData;
+    
+    var CSV = '';    
+    //Set Report title in first row or line
+    
+    CSV += ReportTitle + '\r\n\n';
+
+    //This condition will generate the Label/Header
+    if (ShowLabel) {
+        var row = "";
+        
+        //This loop will extract the label from 1st index of on array
+        for (var index in arrData[0]) {
+            
+            //Now convert each value to string and comma-seprated
+            row += index + ',';
+        }
+
+        row = row.slice(0, -1);
+        
+        //append Label row with line break
+        CSV += row + '\r\n';
+    }
+    
+    //1st loop is to extract each row
+    for (var i = 0; i < arrData.length; i++) {
+        var row = "";
+        
+        //2nd loop will extract each column and convert it in string comma-seprated
+        for (var index in arrData[i]) {
+            row += '"' + arrData[i][index] + '",';
```

Comment:
Have you worked on this issue? This is missing escaping. There is an RFC for implementing propper CSV ([RFC4180](https://www.ietf.org/rfc/rfc4180.txt)). If you absolutely have to implement it yourself follow that one. The python standard library also ships with a perfectly fine [csv writer](https://docs.python.org/3.6/library/csv.html), maybe it would be a better idea to use that one, than rolling your own (client side).